Add NodeWrapper for testing

kaisoz opened the pull request:
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Creating nodes for testing is quite verbose. This PR introduces the `NodeWrapper` to make them their creation more compact.

#### Which issue(s) this PR fixes:
Fixes #3661 

#### Special notes for your reviewer:
This PR only refactors **one test file**. Adding the wrapper and refactoring all test files that create Node objects at once would make the PR too lengthy and difficult to review. The remaining test files will be refactored in a follow-up PR.

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

mimowo commented:
/cc @mbobrovskyi @PBundyra 
for the first review passes

mbobrovskyi commented on pkg/util/testingjobs/node/wrappers.go:
```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```
```suggestion
Copyright 2024 The Kubernetes Authors.
```

PBundyra commented:
Looks good! Please use wrapper in the other parts of code as well, including: 
- `pkg/scheduler/scheduler_test.go`
- `test/integration/controller/jobs/*`
- `test/integration/tas/tas_test.go`

mbobrovskyi reviewed (commented):
Overall LGTM

mimowo commented on pkg/util/testingjobs/node/wrappers.go:
```diff
@@ -0,0 +1,69 @@
+
+// StatusConditions updates status conditions of the Node.
+func (n *NodeWrapper) StatusConditions(conditions ...corev1.NodeCondition) *NodeWrapper {
+	n.Status.Conditions = conditions
```
Maybe append instead of overriding.

mimowo commented on pkg/cache/tas_cache_test.go:
```diff
@@ -48,138 +49,84 @@ func TestFindTopologyAssignment(t *testing.T) {
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+			}).
+			StatusConditions(corev1.NodeCondition{
+				Type:   corev1.NodeReady,
+				Status: corev1.ConditionTrue,
+			}).
```
I think we could also have a helper `Ready()` for more brevity. It will append the condition calling `StatusConditions`. We can still keep the `StatusConditions` function for other uses.

kaisoz replied:
Makes sense!

mimowo commented:
> Looks good! Please use wrapper in the other parts of code as well, including:

I'm also ok to address the other files in a follow up, to de-risk conflicts in the meanwhile, but as you prefer @kaisoz .

kaisoz commented:
> > Looks good! Please use wrapper in the other parts of code as well, including:
> 
> I'm also ok to address the other files in a follow up, to de-risk conflicts in the meanwhile, but as you prefer @kaisoz .

@mimowo I agree with you on that approach (in fact, that's what I proposed in the PR description). I think that changing those files in this same PR will make it too long to be reviewed properly. 

So I refactored the rest of the tests of the `pkg/cache/tas_cache_test.go` file and  updated the PR description 👍🏻

mimowo commented:
/lgtm
/approve
This is a good basis for the follow up (please do) and could already be useful when adding new tests.
